app/models/miq_report_result.rb

```diff
@@ -335,6 +335,19 @@ def self.auto_generated
     where.not(:report_source => "Generated by user")
   end
 
+  def self.with_current_user_groups_and_report(report_id = nil)
+    with_report(report_id).with_current_user_groups
+  end
+
+  def self.with_report(report_id = nil)
+    report_id.nil? ? where.not(:miq_report_id => nil) : where(:miq_report_id => report_id)
+  end
+
+  def self.with_current_user_groups
+    current_user = User.current_user
+    current_user.admin_user? ? all : where(:miq_group_id => current_user.miq_group_ids)
```

Comment:
Note @lpichler, miq_groups_ids was changed a little here, to no longer have a custom method and instead use a rails built-in.  Can you rebase and verify this doesn't adversely affect this PR?